app/models/course.rb

```diff
@@ -49,22 +47,29 @@ class Course < ApplicationRecord
   # this makes use of the shrine gem
   include ScreenshotUploader[:image]
 
-  searchable do
-    text :title
-    integer :program_ids, multiple: true do
-      divisions.pluck(:program_id).uniq
-    end
-    integer :editor_ids, multiple: true
-    boolean :term_independent
-    # this is for ordering
-    string :sort_title do
-      ActiveSupport::Inflector.transliterate(course.title).downcase
-    end
+  include ApplicationHelper
+
+  include PgSearch::Model
+
+  pg_search_scope :search_by_title,
+                  against: [:title, :short_title],
+                  using: {
+                    tsearch: { prefix: true, any_word: true },
+                    trigram: { word_similarity: true,
+                               threshold: 0.3 }
+                  }
+
+  def self.default_search_order
+    Arel.sql("LOWER(unaccent(courses.title))")
   end
```

Comment:
We should be very careful not to include any user-input in these Arel strings as the string in `Arel.sql(...)` is considered to be a _safe_ SQL string, see [the API docs](https://api.rubyonrails.org/classes/Arel.html). You don't use any user input here (even though the course title is indeed a user input, however it is never put inside the sql query as user input; instead it's the _literal_ string `courses.title` that is used here). I just remark this such that we both check this again in detail.

Reply:
I checked again. As you say, since the literal string `courses.title` is used here, there is no security issue here.